sla_per_user_system_test.py

```diff
@@ -0,0 +1,501 @@
+            end_time = time.time()
+
+            self.validate_scheduler_runtime(start_time=start_time, end_time=end_time,
+                                            read_users=read_users, expected_ratio=expected_shares_ratio)
+
+            self.assertTrue(results, msg='Not received cassandra-stress results')
+
+            self.log.debug('Validate cassandra-stress ops deviation')
+            actual_shares_ratio = self.calculate_metrix_ratio_per_user(two_users_list=read_users, metrix=results)
+            self.validate_deviation(expected_ratio=expected_shares_ratio,
+                                    actual_ratio=actual_shares_ratio, msg='Validate cassandra-stress ops.')
+
+        finally:
+            self.clean_auth(entities_list_of_dict=read_users)
+
+    def test_throughput_latency_mixed_load(self):
+        """
+        Test when one user run load with high latency and another  - with high througput
+        The load is run on the full data set (that is read from both the cache and the disk)
+        Troughput - latency test:
+        - Add SLA and grant to user (before any load)
+        - user190 with 190 shares
+        - user950 qith 950 shares
+        - Each user runs load from own loader (round robin):
+           - user190 runs load with throttle
+           - user950 runs load with high throughput
+        """
+        stress_duration = 10  # minutes
+        shares = [190, 950]
+        read_users = []
+
+        session = self.prepare_schema()
+        self.create_test_data()
+
+        # Define Service Levels/Roles/Users
+        for share in shares:
+            read_users.append({'user': User(session=session, name='user%d' % share, password='user%d' % share),
+                               'role': Role(session=session, name='role%d' % share),
+                               'service_level': ServiceLevel(session=session, name='sla%d' % share,
+                                                             service_shares=share)})
+
+        # Create Service Levels/Roles/Users
+        self.create_auths(entities_list_of_dict=read_users)
+
+        # Define stress commands
+        read_cmds = [self.define_read_cassandra_stress_command(user=read_users[0], load_type=self.MIXED_LOAD,
+                                                               workload_type=self.WORKLOAD_THROUGHPUT, threads=200,
+                                                               stress_duration_min=stress_duration),
+                     self.define_read_cassandra_stress_command(user=read_users[1], load_type=self.MIXED_LOAD,
+                                                               workload_type=self.WORKLOAD_LATENCY, threads=250,
+                                                               stress_duration_min=stress_duration)
+                     ]
+
+        self._throughput_latency_tests_run(read_users=read_users, read_cmds=read_cmds)
```

Comment:
Here I think the test scenario should be to run the "latency" load first, to get the latency "before".
Then, we add the "throughput" load and check the latency "after".
If the latency increased too much (DEFINE SOMEWHERE), we should fail the test.

I don't see it here...

Reply:
?

Reply:
Done